actors/util/adt/hamt.go

```diff
@@ -0,0 +1,88 @@
+package adt
+
+import (
+	"context"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-hamt-ipld"
+	typegen "github.com/whyrusleeping/cbor-gen"
+)
+
+type Store interface {
+	Context() context.Context
+	hamt.CborIpldStore
+}
+
+type Keyer interface {
+	Key() string
+}
+
+type HamtStore struct {
```

Comment:
I see the point of this being to abstract many of the HAMT implementation details. To that end, I think we should call this `Map`.